auslib/db.py

```diff
@@ -1040,11 +1060,33 @@ def addRelease(self, name, product, blob, changed_by, transaction=None):
     def updateRelease(self, name, changed_by, old_data_version, product=None, read_only=None, blob=None, transaction=None):
         if product or blob:
             self._proceedIfNotReadOnly(name, transaction=transaction)
+        where = [self.name == name]
         what = {}
-        if read_only is not None:
-            what['read_only'] = read_only
+
+        current_product = self.select(where=where, columns=[self.product], transaction=transaction)[0]["product"]
```

Comment:
It's possible this (and the similar check below) might cause a lot more queries when l10n arrives every morning. Maybe switch to getReleaseBlob() if that becomes a problem (which relies on the blob product == db row product check).